include/hip/hcc_detail/hip_vector_types.h

```diff
@@ -312,6 +312,22 @@ THE SOFTWARE.
 
         using value_type = T;
 
+        __host__ __device__
+        constexpr
+        HIP_vector_base() noexcept : data{} {};
```

Comment:
Can we remove the default ctors ? Since cuda-clang does not allow shared variables to have non-trivial ctors since shared variables cannot be initialized. Adding this ctor will prevent vectors from being used as shared variables.

Reply:
@AlexVlx  do we provide guarantee that it will get zero-initialized by default?

Reply:
nvcc does not initialize int2

https://godbolt.org/z/9HR2MQ